fr.esrf.tango.generator.xtend/src/fr/esrf/tango/pogo/generator/python/PythonUtils.xtend

```diff
@@ -259,33 +259,42 @@ class PythonUtils {
 		«ENDIF»
 		«ENDIF»
 		«IF !cmd.argout.type.voidType»    dtype_out=«cmd.argout.type.pythonTypeHL», 
-		«IF !cmd.argout.description.empty»    doc_out="«cmd.argout.description.oneLineString»", 
-		«ENDIF»
-		«ENDIF»
-		«IF !cmd.description.empty»    description="«cmd.description.oneLineString»",
 		«ENDIF»
     «setAttrPropertyHL("display_level", cmd.displayLevel, false)»
     «setAttrPropertyHL("polling_period", cmd.polledPeriod, false)»
 		«IF cmd.hasCommandArg»    )
 		«ENDIF»
+		«ENDIF»
 	«ENDIF»
-«ENDIF»
     @DebugIt()
     def «cmd.methodName»(self«IF !cmd.argin.type.voidType», argin«ENDIF»):
         «IF cls.description.filestogenerate.toLowerCase.contains("protected regions")»
-            «IF !cmd.argout.type.voidType»
-                «protectedAreaHL(cls, cmd.name, cmd.argout.type.defaultValueReturnHL, false)»
-            «ELSE»
-                pass
-            «ENDIF»
+        «openProtectedAreaHL(cls,cmd.name)»
+        """
+        «cmd.description.commentCmdParamMultiLines»
+        «IF !cmd.argin.type.voidType»
+        
+        :param argin: «cmd.argin.type.pythonTypeHL»
+        «cmd.argin.description.commentCmdParamMultiLines»
+        «ENDIF»
+        «IF !cmd.argout.type.voidType»
+        
+        :return:«cmd.argout.type.pythonTypeHL»
+        «cmd.argout.description.commentCmdParamMultiLines»
         «ELSE»
-            «IF !cmd.argout.type.voidType»
-                return «cmd.argout.type.defaultValueTestHL»
-            «ELSE»
-                pass
-            «ENDIF»
+        
+        :return:None
+        «ENDIF»
+        """
+        «cmd.argout.type.defaultValueReturnHL»
```

Comment:
Not sure what this line is doing?  It is after the docstring, but not being returned.  There was something similar before, but inside `«IF !cmd.argout.type.voidType»`

Reply:
This is return statement of the command, generated as part of command. If command has output argument then the default value of the return type is returned as 'return 0'. If the output argument is void then 'pass' is returned.

Reply:
But there is no `return` on this line.  How does it work?
Line 295 makes sense to me.

Reply:
Because the method 'defaultValueRetunHL' returns the value with 'return' keyword. The method used on line 295 is 'defaultValueTestHL', which return only the value (without 'return' keyword). So both method works.